### ISSUE TITLE
Detect incorrectly formatted value_for_platform usage

### ISSUE THREAD

**Issue by tas50** [chef/cookstyle#442]
### What category of cop is this?:

Please try to select as few as possible.

- [ ] ChefDeprecations - A cop that alerts on a deprecation in the Chef Infra Client
- [ ] ChefCorrectness - A cop that alerts when incorrect coding behavior is being used
- [ ] ChefSharing - A cop that alerts to missing functionality necessary for sharing cookbooks with others
- [ ] ChefStyle - A cop that alerts to a style best practice
- [ ] ChefModernize - A cop that alerts when a cookbook can be simplified or modernized with new functionality
- [ ] ChefEffortless - A cop that alerts on code that must be resolved to move to the Effortless pattern

### Describe the new cop:
A lot of people incorrectly use value_for_platform and don't include the hash after the platform key. We should catch these.

### What it would trigger on:

Bad:
```ruby
value_for_platform(
  'ubuntu' => "http://aerospike.com/download/tools/#{node['aerospike']['version']['tools']}/artifact/ubuntu14"
)
```

Good:
```ruby
value_for_platform(
  'ubuntu' => {
    '~> 14.04' => "http://aerospike.com/download/tools/#{node['aerospike']['version']['tools']}/artifact/ubuntu14",
    'default' => "http://aerospike.com/download/tools/#{node['aerospike']['version']['tools']}/artifact/ubuntu12"
  },
  'debian' => { 'default' => "http://aerospike.com/download/tools/#{node['aerospike']['version']['tools']}/artifact/debian#{node['platform_version']}" },
  %w(amazon centos redhat) => { 'default' => "http://aerospike.com/download/tools/#{node['aerospike']['version']['tools']}/artifact/#{node['aerospike']['package_suffix']}" }
)
```
